shared/app.js

```diff
@@ -92,10 +93,11 @@ module.exports = Backbone.Model.extend({
       this.templateAdapter = this.options.templateAdapterInstance;
     } else {
       var templateAdapterModule = attributes.templateAdapter || this.defaults.templateAdapter,
-      templateAdapterOptions = {entryPath: entryPath};
+        templateAdapterOptions = {entryPath: entryPath},
+        templateEngine = require(this.defaults.templateEngine);
```

Comment:
i think `attributes.templateEngine || this.defaults.templateEngine` makes sense, although I think that if you have a defaults object backbone will do that automatically for you.  If you have `this.attributes.templateEngine` the defaults will be set properly to that attribute if nothing is provided. http://backbonejs.org/#Model-defaults